module/Company/src/Company/Controller/AdminController.php

```diff
@@ -193,6 +199,75 @@ public function addJobAction()
     }
 
     /**
+     * Action that displays a form for editing a category
+     *
+     *
+     */
+    public function editCategoryAction()
+    {
+        // Get useful stuff
+        $companyService = $this->getCompanyService();
+        $categoryForm = $companyService->getCategoryForm();
+
+        // Get parameter
+        $categoryId = $this->params('categoryID');
+
+        // Get the specified company
+        $categories = $companyService->getAllCategoriesById($categoryId);
+
+        // If the company is not found, throw 404
+        if (empty($categories)) {
+            $company = null;
+            return $this->notFoundAction();
+        }
+
+        // Handle incoming form data
+        $request = $this->getRequest();
+        if ($request->isPost()) {
+            if ($companyService->saveCategory(
+                $request->getPost()
+            )){
+                return $this->redirect()->toRoute(
+                    'admin_company/editCategory',
+                    [
+                        'categoryID' => $categoryId,
+                    ],
+                    [],
+                    false
```

Comment:
FYI, these last two parameters are optional and can be committed, I believe `[]` and `false` are their default values.